### ISSUE TITLE
Dataset: NQuads serialization only makes sense for context-aware stores!

### ISSUE THREAD

**Issue by sdasda7777** [RDFLib/rdflib#2335]
Hi, I'd like to serialize JSON-LD to N-Quads, however I keep getting `NQuads serialization only makes sense for context-aware stores!` exception. From the Dataset definition and [this issue](https://github.com/RDFLib/rdflib/issues/1699) it seems that it however should work, correct? Here is my code (formatting might be bit off):

```Python
from rdflib import Dataset
        
    try:
        data = """{
  "@context": "http://schema.org/",
  "@type": "Person",
  "name": "Jane Doe",
  "jobTitle": "Professor",
  "telephone": "(425) 123-4567",
  "url": "http://www.janedoe.com"
}"""
         ds = Dataset()
         ds = ds.parse(data=data, format="json-ld")
         print(ds.serialize(format="nquads"))
    except Exception as e:
         print(e)
```

**Comment by aucampia** [comment]
@sdasda7777 this is a known bug (IMO at least) recorded here <https://github.com/RDFLib/rdflib/issues/1892>.

I will close this as a duplicate, however, I think that NQuads for a non-context-aware store should be identical to NTriples [[ref](https://www.w3.org/TR/n-quads/#simple-triples)].

**Comment by sdasda7777** [comment]
> @sdasda7777 this is a known bug (IMO at least) recorded here #1892.

I see, I was hoping for some kind of solution, but since it's open for that long I guess I can forget about that.

> I think that NQuads for a non-context-aware store should be identical to NTriples.

For this example, yes. However JSON-LD may actually have context information, and I want it to work in that case as well.

**Comment by sdasda7777** [comment]
[Section number 8](https://www.w3.org/TR/json-ld11/#data-model) of the standard has an example that should translate into multiple graphs. You can see it really does when you try converting it to N-Quads on the JSON-LD playground, which is exactly what I want to be able to do.

**Comment by aucampia** [comment]
> For this example, yes. However JSON-LD may actually have context information, and I want it to work in that case as well.

But then you should use a context-aware store like `rdflib.graph.Dataset`, otherwise it will be lost anyway.

> I see, I was hoping for some kind of solution, but since it's open for that long I guess I can forget about that.

The reason this bug is not fixed, even though it is simple to fix, is that it can easily be worked around by users, even if we fix it, it will be no different from just using NTriples, and this is because non-context aware stores cannot contain named graphs, and if there are no graph names NQuads is equivalent to NTriples.

We will be grateful for PRs that fix <https://github.com/RDFLib/rdflib/issues/1892>.

**Comment by sdasda7777** [comment]
> But then you should use a context-aware store like rdflib.graph.Dataset, otherwise it will be lost anyway.

But I am using a Dataset, am I not?

> The reason this bug is not fixed, even though it is simple to fix, is that it can easily be worked around by users, even if we fix it, it will be no different from just using NTriples, and this is because non-context aware stores cannot contain named graphs, and if there are no graph names NQuads is equivalent to NTriples.

Okay, so I should be able to do this conversion just fine if I use context-aware store? Which stores are those?

**Comment by sdasda7777** [comment]
> Dataset should be context aware, but I guess for some reason the serializers disagree. [rdflib.graph.ConjunctiveGraph](https://rdflib.readthedocs.io/en/stable/apidocs/rdflib.html#rdflib.graph.ConjunctiveGraph) is also context aware, and I suspect that should work, could you try using that instead of rdflib.graph.Dataset ?

Yeah, it is the same output, `NQuads serialization only makes sense for context-aware stores!` as well.

I am using Python 3.10.4 and rdflib 6.3.2, if that helps in any way.

**Comment by aucampia** [comment]
Okay, then I don't really have an immediate answer. I will look at it when I have time, but it does seem like a bug, but a pull requests fixing this will be welcome.

**Comment by sdasda7777** [comment]
Oh, you're right! Thank you so much!!!

**Comment by namedgraph** [comment]
@aucampia how about making `nquads` the default serialization for `Dataset`? Currently `ds.serialize()` returns nothing because I'm assuming it defaults to `ntriples`? Should I create a separate issue for this?

**Comment by aucampia** [comment]
@namedgraph there is actually a PR that will fix this - though it will use `trig` instead of `nquads`:

https://github.com/RDFLib/rdflib/pull/2373

But this needs some other things fixed before we can merge it.

**Comment by aucampia** [comment]
The reason the output is empty is because of this bug: 

- https://github.com/RDFLib/rdflib/issues/2393
